src/commands/cmd_tdigest.cc

```diff
@@ -0,0 +1,149 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+#include "command_parser.h"
+#include "commander.h"
+#include "server/redis_reply.h"
+#include "server/server.h"
+#include "status.h"
+#include "string_util.h"
+#include "types/redis_tdigest.h"
+
+namespace redis {
+namespace {
+constexpr auto kCompressionArg = "compression";
+constexpr auto kErrWrongKeyword = "T-Digest: wrong keyword";
+constexpr auto kErrWrongNumOfCreateArguments = "wrong number of arguments for 'tdigest.create' command";
+constexpr auto kErrWrongNumOfInfoArguments = "wrong number of arguments for 'tdigest.info' command";
+constexpr auto kErrParseCompression = "T-Digest: error parsing compression parameter";
+constexpr auto kErrCompressionMustBePositive = "T-Digest: compression parameter needs to be a positive integer";
+const static auto kErrCompressionOutOfRange =
+    "T-Digest: compression must be between 1 and " + std::to_string(kTDigestMaxCompression);
+constexpr auto kErrKeyNotFound = "T-Digest: key does not exist";
+constexpr auto kErrKeyAlreadyExists = "T-Digest: key already exists";
+
+constexpr auto kInfoCompression = "Compression";
+constexpr auto kInfoCapacity = "Capacity";
+constexpr auto kInfoMergedNodes = "Merged nodes";
+constexpr auto kInfoUnmergedNodes = "Unmerged nodes";
+constexpr auto kInfoMergedWeight = "Merged weight";
+constexpr auto kInfoUnmergedWeight = "Unmerged weight";
+constexpr auto kInfoObservations = "Observations";
+constexpr auto kInfoTotalCompressions = "Total compressions";
+constexpr auto kInfoMemoryUsage = "Memory usage";
+}  // namespace
+
+class CommandTDigestCreate : public Commander {
+ public:
+  Status Parse(const std::vector<std::string> &args) override {
+    CommandParser parser(args, 1);
+    key_name_ = GET_OR_RET(parser.TakeStr());
+    options_.compression = 100;
+    bool invalid_keyword = false;
+    if (args.size() != 2 && (args.size() != 4 || (invalid_keyword = !util::EqualICase(kCompressionArg, args[2])))) {
+      return {Status::RedisParseErr, invalid_keyword ? kErrWrongKeyword : kErrWrongNumOfCreateArguments};
+    }
+    if (parser.EatEqICase(kCompressionArg)) {
+      auto status = parser.TakeInt<int32_t>();
+      if (!status) {
+        return {Status::RedisParseErr, kErrParseCompression};
+      }
+      auto compression = *status;
+      if (compression <= 0) {
+        return {Status::RedisParseErr, kErrCompressionMustBePositive};
+      }
+      if (compression < 1 || compression > static_cast<int32_t>(kTDigestMaxCompression)) {
+        return {Status::RedisParseErr, kErrCompressionOutOfRange};
+      }
+      options_.compression = static_cast<uint32_t>(compression);
+    }
```

Comment:
```suggestion
    if (parser.EatEqICase(kCompressionArg)) {
      auto status = parser.TakeInt<int32_t>();
      if (!status) {
        return {Status::RedisParseErr, kErrParseCompression};
      }
      auto compression = *status;
      if (compression <= 0) {
        return {Status::RedisParseErr, kErrCompressionMustBePositive};
      }
      if (compression < 1 || compression > static_cast<int32_t>(kTDigestMaxCompression)) {
        return {Status::RedisParseErr, kErrCompressionOutOfRange};
      }
      options_.compression = static_cast<uint32_t>(compression);
    }
    if (parser.Good()) {
      return ... // invalid arguments
    }
```

Reply:
Hi @PragmaTwice ,

Thanks for your advice! 😊

I have refactored the block with `parser.Good()`.

Best Regards,
Edward